[libcxx][NFC] Consolidate testing concept CanBePiped

hawkinsw opened the pull request:
Almost every test needed a CanBePiped concept and each implemented it separately, but identically. Consolidate all implementations into test_range.h.

hawkinsw committed:
[libcxx][NFC] Consolidate testing concept CanBePiped

Almost every test needed a CanBePiped concept and each implemented it
separately, but identically. Consolidate all implementations into
test_range.h.

hawkinsw commented:
I hope that this consolidation is helpful!

hawkinsw commented:
@cjdb I hope that this is something that is helpful!

hawkinsw commented:
Hello! Let me know if there is anything else I can do to get this into shape! cc @cjdb @var-const

var-const commented on libcxx/test/std/ranges/range.adaptors/range.transform/adaptor.pass.cpp:
```diff
@@ -20,11 +20,6 @@
 #include "test_macros.h"
```
Do you need to include `"test_range.h"` here? Otherwise, this file is probably relying on a transitive include (unless `CanBePiped` is not used here?).

hawkinsw replied:
Great eyes! In these cases, I was relying on the fact that the implementers of the adapters created their own header files (`types.h`) and those file includes the `test_range.h`. I have adjusted the patch to make direct inclusion of the `test_ranges.h` file but I am more than happy to go back to the old version if that is our standard.

Thanks for the sharp feedback!

var-const commented on libcxx/test/std/ranges/range.adaptors/range.take.while/adaptor.pass.cpp:
```diff
@@ -42,12 +42,6 @@ static_assert(std::is_invocable_v<decltype((std::views::take_while)), int (&)[2]
-      { std::forward<View>(view) | std::forward<T>(t) };
-    };
-
 static_assert(!CanBePiped<MoveOnlyView, decltype(std::views::take_while)>);
```
Include `"test_range.h"`? (or is it not used in this file?)

var-const reviewed (commented):
Thanks, this is a nice refactoring! Just wanted to double-check whether a couple of these files need to include `test_range.h` now (or perhaps they aren't actually using `CanBePiped`?). Other than that, LGTM.

hawkinsw committed:
fixup! [libcxx][NFC] Consolidate testing concept CanBePiped

Directly import test_range.h in several places.

hawkinsw commented:
Just wanted to see if my replies addressed your comments! Thanks for everything!
Will

hawkinsw commented:
> Just wanted to see if my replies addressed your comments! Thanks for everything! Will

Please let me know if there is anything else I can do to clean this up!

var-const reviewed (approved):
Sorry about the delay! LGTM.
